### ISSUE TITLE
Few questions regarding this project

### ISSUE THREAD

**Issue by Joe23232** [UPBGE/upbge#1326]
Hi I have a few questions regarding this project.

1. Since it is under GPL license, do I have to legally disclose my source code for the game that I produce?
2. Am I able to use Rust for this by any chance? If so I hope there is not too much _friction_ when using Rust?
3. Is this an extension or a fork of Blender? If it is a fork of Blender, wouldn't it make more sense to have it as an extension?
4. Is there some ECS integration or something? Or is this something planned in the near future?

**Comment by lordloki** [comment]
Hi, 
1a. If you export your game with the blenderplayer separated from blend files then the blenderplayer will be under GPL, your scripts that use bpy or bge will be under GPL but your models, textures, blends with logic bricks, sound, etc will be under your own license. 
1b. If your export all your game under a bundle executable (using for it save as game engine addon) the all your code and assets will be under GPL.
1c. If you export your game using for it the save as game engine addon but the bundle excutable only includes a little blend file that call to another external blend file using the game actuator logic brick, then the executable plus the "by-pass" blend will be under the GPL, your scripts that use bpy or bge will be under GPL but your models, textures, blends with logic bricks, sound, etc will be under your own license. 

2. For upbge you can use mainly python although there are several users that used another scripting language like rust but i think that they some kind the binding for python. Also, you can program in C++ modifying directly the sources (for adding a new python binding or a new logic brick or whatever)

3. it is a fork of Blender. We can't have it as extension because we modify the DNA and RNA sources of blender. This way we can do the blend format compatible from Blender to UPBGE without any problem.

4. I don't know exactly whar are you refferering? do you mean to integrate it with Amazon ECS or system like that?

In summary, the software and source code are bound to the GNU GPL, but the blend-files (models, textures, sounds) are not.

Standalone Games
In case you save out your game as a single standalone (using addons for this purpose, for example), the blend-file gets included in the binary (the Blender Player). That requires the blend-file to be compatible with the GNU GPL license.

In this case, you could decide to load and run another blend-file game (using the Game Actuator logic brick). That file then is not part of the binary, so you can apply any license you wish on it.

**Comment by Joe23232** [comment]
THanks for your response :)

> 1a. If you export your game with the blenderplayer separated from blend files then the blenderplayer will be under GPL, your scripts that use bpy or bge will be under GPL but your models, textures, blends with logic bricks, sound, etc will be under your own license.
1b. If your export all your game under a bundle executable (using for it save as game engine addon) then all your code and assets will be under GPL.
1c. If you export your game using for it the save as game engine addon but the bundle excutable only includes a little blend file that call to another external blend file using the game actuator logic brick, then the executable plus the "by-pass" blend will be under the GPL, your scripts that use bpy or bge will be under GPL but your models, textures, blends with logic bricks, sound, etc will be under your own license.

Isn't there a way I can use scripts and not have it under GPL license? What if I used Rust? Because honestly this drives a lot of potential game developers away from using this game engine no offense.

> although there are several users that used another scripting language like rust but i think that they some kind the binding for python.

I see, is there a link to Rust bindings or something?

> I don't know exactly what are you referring? do you mean to integrate it with Amazon ECS or system like that?

Have you heard of game engines such as Bevy? It has its own ECS, there are other ECS such as shipyard and legion.

**Comment by lordloki** [comment]
> Isn't there a way I can use scripts and not have it under GPL license? What if I used Rust? Because honestly this drives a lot of potential game developers away from using this game engine if we can't close source our own game we produce no offense.
You can uses a scripts under a non-GPL license if you don't use the bge or bpy apis. If you use them then your script will be GPL.

You can avoid the GPL creating a server-client game and to have all the logic in the server. Another way is using logic bricks and encrypt them with BPPlayer.

There is no way that we can change the GPL license as it is the Blender license and we are a fork of Blender. We can not do anything about it. If this is an unavoidable problem for any game developer then they have to use another game engine more "free".

> I see, is there a link to Rust bindings or something?

There is no official (and no official) links to Rust bindings or another scripting system. There are only python bindings. The users that uses Rust they do it privately. I haven't seen any Rust binding code.

Regarding to ECS it is not in our planning.

**Comment by Joe23232** [comment]
Ah I see, thansk anyways :)

**Comment by BluePrintRandom** [comment]
@Joe23232 regarding the API - they have everything and the kitchen sink wrapped in py,

saying "I want to use your free stuff but not give away my glue using it" - is a jerk move,

GameDev can mature like science.
if there is some key feature you think is amazing and work money - open a kickstart or a gofund me and ask for the money before you release it
